### DISCUSSION TITLE
Client VMs invisible in the workload/app graph view but visible in the workload's "Graph Overview"

### DISCUSSION THREAD

**Discussion by spockz** [kiali/kiali#3692] [Q&A] [answered]
Our VMs that *only consume* services are invisible in the main Graph views. VMs *receiving* traffic however are visible. When I go to the page of a workload or service that is being consumed by a VM the VM *does* show up. See screenshots.
<img width="1500" alt="Screenshot 2021-02-11 at 12 15 51" src="https://user-images.githubusercontent.com/54579/107630147-6eb74a00-6c63-11eb-8701-42891f538314.png">
<img width="1535" alt="Screenshot 2021-02-11 at 12 15 38" src="https://user-images.githubusercontent.com/54579/107630155-7119a400-6c63-11eb-8b93-044055a984af.png">
<img width="1662" alt="Screenshot 2021-02-11 at 12 15 28" src="https://user-images.githubusercontent.com/54579/107630158-724ad100-6c63-11eb-934d-d8b35f5d669d.png">

**Comment by lucasponce** [comment, marked as answer]
I think this could be related to the source/destination telemetry discussion that there are already in place.

@spockz, if possible, could you give a try to this PR https://github.com/kiali/kiali/pull/3680 I know that @jshaughn is working on this area and I'm curious if this approach may help on your scenario.

**Reply by jshaughn** [reply]
+1, this may improve your results.  Please let us know if you can test the PR.  Otherwise it may be part of the v1.30 release if it passes review and QE.

**Reply by israel-hdez** [reply]
@jshaughn I find weird that the mini-graph is showing the node that is missing in the main graph. Perhaps there is a difference between queries in main vs drilled down graph?

**Reply by jshaughn** [reply]
There is a difference in that the node detail graphs are generated wrt to the node, so incoming is based on dest telemetry and outgoing on source telemetry.

**Reply by spockz** [reply]
I have tried the build from the PR with the docker image provided in the dev channel and our server VMs disappear from the graph. Using the docker image with tag 1.30.0 from the official restores the display of that VM again.

**Reply by jshaughn** [reply]
Are you saying that the PR did not resolve the issue, but that v1.30 did resolve the issue?  That seems backwards :)

**Reply by spockz** [reply]
Not really. I'm saying that with 1.30.0 we get the same behaviour as with 1.29 where we were able to see VM *servers*, receiving traffic, but not the VM *clients*, generating traffic. With the image from the PR none of the VMs were visible.

**Reply by jshaughn** [reply]
OK, it makes sense that 1.29 and 1.30 would be the same, as not change was made.  It seems very strange that you would lose visualization with the PR, as opposed to gain.  I can't really explain that result.

**Reply by jshaughn** [reply]
Well, after more detailed investigation it became clear why things were not showing up as desired.  A new revision of the PR has been generated and tested, and is looking better.

**Reply by jshaughn** [reply]
FYI: the final revision of that PR work will be in Kiali 1.32.
